state/resources_state_resource.go

```diff
@@ -75,10 +78,11 @@ type resourceStorage interface {
 }
 
 type resourceState struct {
-	persist resourcePersistence
-	raw     rawState
-	storage resourceStorage
-	clock   clock.Clock
+	persist               resourcePersistence
+	raw                   rawState
+	dockerMetadataStorage DockerResourceStorage
```

Comment:
This makes me think DockerMetadataStorage / DockerMetadata etc is a better name for this stuff? Resource is more a Juju specific term but what we are really doing is storing metadata used to access a docker image